realm/realm-library/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -29,16 +33,27 @@
  * Class that wraps a normal RealmObject in order to allow dynamic access instead of a typed interface.
  * Using a DynamicRealmObject is slower than using the regular RealmObject class.
  */
-public final class DynamicRealmObject extends RealmObject {
+public final class DynamicRealmObject extends RealmObject implements RealmObjectProxy {
     private String className;
 
+    // TODO This fields is duplicated between DynamicRealmObject and Proxy classes....eeew
+    // Brainstorm idea: Composition over inheritance. Perhaps encapsulate in a RealmObjectProxyState class ?
```

Comment:
Agree here. Composition > Inheritance. I'd propose the name `ProxyState` because the alternative just seems too long.